### PR TITLE
Release 4.1

### PR DESCRIPTION
**Pull request by ggabernet**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/airrflow/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/airrflow _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nf-test test main.nf.test -profile test,docker`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow_schema.json [hunk 1]
```diff
@@ -742,6 +742,13 @@
                     "description": "Validation of parameters in lenient more.",
                     "hidden": true,
                     "help_text": "Allows string values that are parseable as numbers or booleans. For further information see [JSONSchema docs](https://github.com/everit-org/json-schema#lenient-mode)."
+                },
+                "pipelines_testdata_base_path": {
+                    "type": "string",
+                    "fa_icon": "far fa-check-circle",
+                    "description": "Base URL or local path to location of pipeline test dataset files",
+                    "default": "https://raw.githubusercontent.com/nf-core/test-datasets/",
```
Comment (maxulysse) [inline nextflow_schema.json:750]:
```suggestion
                    "default": "https://raw.githubusercontent.com/nf-core/test-datasets/airflow/",
```

File: nextflow.config [hunk 2]
```diff
@@ -14,7 +14,7 @@ params {
     mode = "fastq"
     miairr="$projectDir/assets/reveal/mapping_MiAIRR_BioSample_v1.3.1.tsv"
     index_file = false
-
+    pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/'
```
Comment (maxulysse) [inline nextflow.config:17]:
```suggestion
    pipelines_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/airflow/'
```

File: nextflow.config [hunk 3]
```diff
@@ -81,8 +81,8 @@ params {
     // -----------------------
     productive_only = true
     reassign = true
-    reference_igblast = 'https://raw.githubusercontent.com/nf-core/test-datasets/airrflow/database-cache/igblast_base.zip'
-    reference_fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/airrflow/database-cache/imgtdb_base.zip'
+    reference_igblast = "${params.pipelines_testdata_base_path}airrflow/database-cache/igblast_base.zip"
+    reference_fasta = "${params.pipelines_testdata_base_path}airrflow/database-cache/imgtdb_base.zip"
```
Comment (maxulysse) [inline nextflow.config:85]:
```suggestion
    reference_igblast = "${params.pipelines_testdata_base_path}database-cache/igblast_base.zip"
    reference_fasta = "${params.pipelines_testdata_base_path}database-cache/imgtdb_base.zip"
```

**Review by maxulysse — APPROVED** [review]
Minor suggestions about what I think should be the proper usage for this params

File: nextflow.config (hunk 2 above)
Comment (JoseEspinosa) [reply nextflow.config:17]: Just as a warning, if this change is applied then `airflow` would need to be removed from all the test configs files

File: docs/usage.md [hunk 4]
```diff
@@ -230,18 +228,29 @@ This profile executes the commands based on the pRESTO pre-set pipeline [presto-
 - Align and annotate the internal C Region (for the BCR specific protocol) for a more specific isotype annotation.
 - Remove duplicate sequences and filter to sequences with at least 2 supporting sources.
 
-Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primers:
+Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primers. Here an example using the mouse IG primers on the Immcantation Bitbucket repository:
```
Comment (JoseEspinosa) [inline docs/usage.md:231]:
```suggestion
Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primers. Here is an example using the mouse IG primers from the Immcantation Bitbucket repository:
```

File: docs/usage.md [hunk 5]
```diff
@@ -266,13 +275,23 @@ This profile executes the sequence assembly commands based on the pRESTO pre-set
 
 After the sequence assembly steps, the remaining steps are common for all protocols.
 
-Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primer sequences:
+Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primer sequences. Here an example using the mouse IG primers on the Immcantation Bitbucket repository:
```
Comment (JoseEspinosa) [inline docs/usage.md:278]:
```suggestion
Please note that the default primer sequences and internal CRegion sequences are for human. If you wish to run this protocol on mouse or other species, please provide the alternative primer sequences. Here is an example using the mouse IG primers from the Immcantation Bitbucket repository:
```

File: modules/local/airrflow_report/airrflow_report.nf [hunk 6]
```diff
@@ -6,8 +6,8 @@ process AIRRFLOW_REPORT {
         error "nf-core/airrflow currently does not support Conda. Please use a container profile instead."
     }
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
```
Comment (JoseEspinosa) [inline modules/local/airrflow_report/airrflow_report.nf:8]: I would simplify this to the suggestion below, as actually is the same image, but not a strong opinion (if agree also do it for the rest of modules).
```suggestion
    container "docker.io/immcantation/airrflow:4.1.0"
```

File: modules/local/airrflow_report/airrflow_report.nf [hunk 7]
```diff
@@ -6,8 +6,8 @@ process AIRRFLOW_REPORT {
...
-        'docker.io/immcantation/airrflow:4.0.0':
-        'docker.io/immcantation/airrflow:4.0.0' }"
+        'docker.io/immcantation/airrflow:4.1.0':
+        'docker.io/immcantation/airrflow:4.1.0' }"
```
Comment (JoseEspinosa) [inline modules/local/airrflow_report/airrflow_report.nf:10]: See my previous comment
```suggestion
```

**Review by JoseEspinosa — APPROVED** [review]
Awesome! 👍  
Only minor comments and it seems that one test is failing, feel free to merge after addressing them 🚀

File: nextflow.config (hunk 2 above)
Comment (ggabernet) [reply nextflow.config]: Thanks, I've updated both accordingly!

**Comment by ggabernet** [comment]
> Awesome! 👍 Only minor comments and it seems that one test is failing, feel free to merge after addressing them 🚀

Not sure why the download workflow pull_request_target is failing, but the `pull_request` one is passing! So I think we're good to go

**Pull request merged**